omnigibson/simulator.py

```diff
@@ -373,6 +380,11 @@ def _non_physics_step(self):
         if len(self._objects_to_initialize) > 0 and self.is_playing():
             for obj in self._objects_to_initialize:
                 obj.initialize()
+                if len(obj.states.keys() & self.object_state_types_on_contact) > 0:
+                    self._objects_require_contact_callback = True
```

Comment:
this is cool! We should probably also make sure these flags get reset every time a new stage is opened / sim is cleared. (actually, in general, maybe put a bunch of stuff in an explicit function like `reset_variables` or something)

Reply:
done